src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter(person => person.sex === 'f'
+    && people.find(x => x.mother === person.name))
+    : people.filter(person => person.sex === 'f');
+
+  const ages = women.map(person => person.died - person.born);
+  const sum = ages.reduce((accumulator, age) => accumulator + age, 0);
+  const average = sum / ages.length;
```

Comment:
You can move this logic to a separate function and use it for `men` and `women` calculations.